common/display/vblankeventhandler.cpp

```diff
@@ -87,12 +97,19 @@ void VblankEventHandler::HandlePageFlipEvent(unsigned int sec,
   int64_t timestamp = ((int64_t)sec * kOneSecondNs) + ((int64_t)usec * 1000);
   IPAGEFLIPEVENTTRACE("HandleVblankCallBack Frame Time %f",
                       static_cast<float>(timestamp - last_timestamp_) / (1000));
+  double vperiod = static_cast<double>(timestamp - last_timestamp_) / (1000));
   last_timestamp_ = timestamp;
 
   IPAGEFLIPEVENTTRACE("Callback called from HandlePageFlipEvent. %lu",
                       timestamp);
   spin_lock_.lock();
   if (enabled_ && callback_) {
+     if(abs(vperiod - vperiod_) > (1/90 - 1/75))
+      callback_2_4_->Callback(display_, timestamp, static_cast<uint32_t>(vperiod * 1000 * 1000));
+     else
+      callback_->Callback(display_, timestamp);
```

Comment:
need to check if the original Callback still need to be invoked, or it is replaced by the new one.